### ISSUE TITLE
Don't open new output file when restarting if old output file isn't full

### ISSUE THREAD

**Issue by PeterCaldwell** [E3SM-Project/scream#2200]
Currently, output files that are supposed to have 8 entries but only fill 4 before a restart (for example) will start the next job submission with a new output file. We should instead re-open the file and fill the remaining 4 entries before creating a new file.

**Comment by AaronDonahue** [comment]
@bartgol , is this issue basically solved?  Can we close?

**Comment by bartgol** [comment]
I'm not entirely sure. I did implement the possibility to resume an existing file, but we did have some hangs happening, so we implemented the `force_new_file` option, which made restarts ignore the existing file (effectively invalidating the resume feature).

I don't know if we are still using that option, perhaps @ndkeen or @crterai know. If we are not (and therefore we don't experience hangs anymore), then we can consider this solved. If we still use the workaround though, we need to investigate.
